bundles/org.openhab.binding.nibeheatpump/src/main/resources/OH-INF/thing/SMO40-types.xml

```diff
@@ -0,0 +1,9437 @@
+	</channel-type>
+	<channel-type id="smo40-48284" advanced="true">
+		<item-type>Switch</item-type>
+		<label>SG Ready hot water</label>
+		<description>Sets whether or not SG Ready should affect hot water</description>
+		<state readOnly="false"></state>
+	</channel-type>
+	<channel-type id="smo40-48285" advanced="true">
+		<item-type>Switch</item-type>
+		<label>SG Ready pool</label>
+		<description>Sets whether or not SG Ready should affect pool</description>
+		<state readOnly="false"></state>
+	</channel-type>
+	<channel-type id="smo40-48367" advanced="true">
+		<item-type>Number</item-type>
+		<label>Speed circ.pump Heat Slave 8</label>
+		<description></description>
+		<state min="1" max="100" step="1" pattern="%d %" readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="smo40-48368" advanced="true">
+		<item-type>Number</item-type>
+		<label>Speed circ.pump Heat Slave 7</label>
+		<description></description>
+		<state min="1" max="100" step="1" pattern="%d %" readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="smo40-48369" advanced="true">
+		<item-type>Number</item-type>
+		<label>Speed circ.pump Heat Slave 6</label>
+		<description></description>
+		<state min="1" max="100" step="1" pattern="%d %" readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="smo40-48370" advanced="true">
+		<item-type>Number</item-type>
+		<label>Speed circ.pump Heat Slave 5</label>
+		<description></description>
+		<state min="1" max="100" step="1" pattern="%d %" readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="smo40-48371" advanced="true">
+		<item-type>Number</item-type>
+		<label>Speed circ.pump Heat Slave 4</label>
+		<description></description>
+		<state min="1" max="100" step="1" pattern="%d %" readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="smo40-48372" advanced="true">
+		<item-type>Number</item-type>
+		<label>Speed circ.pump Heat Slave 3</label>
+		<description></description>
+		<state min="1" max="100" step="1" pattern="%d %" readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="smo40-48373" advanced="true">
+		<item-type>Number</item-type>
+		<label>Speed circ.pump Heat Slave 2</label>
+		<description></description>
+		<state min="1" max="100" step="1" pattern="%d %" readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="smo40-48374" advanced="true">
+		<item-type>Number</item-type>
+		<label>Speed circ.pump Heat Slave 1</label>
+		<description></description>
+		<state min="1" max="100" step="1" pattern="%d %" readOnly="false">
+		</state>
+	</channel-type>
+	<channel-type id="smo40-48375" advanced="true">
+		<item-type>Switch</item-type>
+		<label>Op mode circ.pump Heat Slave 8</label>
+		<description></description>
+		<state readOnly="false"></state>
+	</channel-type>
+	<channel-type id="smo40-48376" advanced="true">
+		<item-type>Switch</item-type>
+		<label>Op mode circ.pump Heat Slave 7</label>
+		<description></descr
```

Comment:
Is this file generated code? Actually labels are expected to be as short as possible. Guideline is 2-3 words with up to 25 chars. See https://www.openhab.org/docs/developer/bindings/thing-xml.html#formatting-labels-and-descriptions

Reply:
As written before, the strings are given by the manufacturer and I won't change them.